src/IO/WriteBuffer.h

```diff
@@ -42,7 +42,8 @@ class WriteBuffer : public BufferBase
     {
         if (!offset())
             return;
-        bytes += offset();
```

Comment:
`bytes += offset()` seems more clear

Reply:
I have noticed in that task some minor bug.

Here is an explanation:
https://github.com/ClickHouse/clickhouse-private/issues/2767#issuecomment-1404291410

> It leads to incorrect count() inside nextImpl(). There offset() is count twice. But I have not found any significant cases when count is called inside nextImpl(). Only for pregress bar in ReadWriteBufferFromHTTPBase for next_callback, it is progress tracker as I see.
For readBuffer problem the same.

May be you wish a comment here instead reversing the code?

Reply:
I don't understand how it could cause offset() to be counted twice. Can you explain that a little bit?

Reply:
Yes, sure :)

Look at this: https://github.com/ClickHouse/ClickHouse/blob/212aef3dc21ce2b30b0a86b8e67ce7a0818bfebf/src/IO/BufferBase.h#L92

And after at this:
https://github.com/ClickHouse/ClickHouse/blob/212aef3dc21ce2b30b0a86b8e67ce7a0818bfebf/src/IO/WriteBuffer.h#L45-L50

If you call count() inside nextImpl() it add offset() twice. I step into it and notice that. By a lucky chance CH doesn't call count inside nextImpl except minor cases like that:
https://github.com/ClickHouse/ClickHouse/blob/212aef3dc21ce2b30b0a86b8e67ce7a0818bfebf/src/IO/ReadWriteBufferFromHTTP.cpp#L443